internal/impl/openai/speech_processor.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openai
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"slices"
+
+	oai "github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai"
+	"github.com/redpanda-data/benthos/v4/public/bloblang"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	ospFieldInput          = "input"
+	ospFieldVoice          = "voice"
+	ospFieldResponseFormat = "response_format"
+)
+
+func init() {
+	err := service.RegisterProcessor(
+		"openai_speech",
+		speechProcessorConfig(),
+		makeSpeechProcessor,
+	)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func speechProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("AI").
+		Summary("Processor that uses the OpenAI API to generate audio from the input text.").
+		Description(`
+This processor calls the OpenAI API for each message, creating speech. By default the entire message's payload as a string is submitted, and the `+"`"+ospFieldInput+"`"+` configuration field allows customizing that.
```

Comment:
This processor sends a text description and other attributes, such as a voice type and format, to the OpenAI API, which generates audio. By default, the processor submits the entire payload of each message as a string, unless you use the `+"`"+oipFieldPrompt+"`"+` configuration field to customize it.